packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuSplitButton.types.ts

```diff
@@ -0,0 +1,94 @@
+
+import { IContextualMenuItem } from '../../ContextualMenu';
+import { IMenuItemClassNames } from './ContextualMenu.classNames';
+import { IContextualMenuItemProps } from './ContextualMenuItem.types';
+import { ContextualMenuSplitButton } from './ContextualMenuSplitButton';
+
+export interface IContextualMenuSplitButtonProps extends React.Props<ContextualMenuSplitButton> {
+  /**
+   * Optional callback to access the ContextualmenuSplitButton interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: ContextualMenuSplitButton | null) => void;
+
+  /**
+   * The item that is used to render the split button.
+   */
+  item: IContextualMenuItem;
+
+  /**
+   * CSS class to apply to the context menu.
+   */
+  classNames: IMenuItemClassNames;
```

Comment:
Hmm. This is not the recommended pattern; could you change to `getStyles`? Or is there a reason you're doing this?

The problem with classNames is that it can't be a function of the state.

Reply:
The only reason I did it this way was because I was passing an existing style, but since we're separating the split button from the menu, I think it makes sense to separate out the styles here.

Reply:
that is totally fine... fwiw: you can still have getStyles, which takes those in.

```
getStyles={ props => { icon: className.icon, checkmarkIcon: classNames.checkmarkIcon }
```

Reply:
Huh, I see, so add it an as an addition got it. I'll do something like this for my next iteration on this.